### ISSUE TITLE
Regularization question

### ISSUE THREAD

**Issue by ashnair1** [facebookresearch/detectron2#2127]
Just wanted to confirm something regarding #239. detectron2 does not explicitly add  a regularisation term to the loss right? That is to say, you use the weight decay parameter in the SGD optimizer to function as your regulariser correct?

I'm curious because while weight decay and L2 regularisation are equivalent while using vanilla SGD, they're not while using SGD with momentum or other optimisers. So does detectron2 (or I guess pytorch) go with weight_decay as the de facto means of regularisation?

**Comment by ppwwyyxx** [comment]
Yes. This is the de facto approach in pytorch.
